pkgs/development/libraries/quarto/default.nix

```
@@ -0,0 +1,63 @@
+{ stdenv
+, lib
+, pandoc
+, esbuild
+, deno
+, fetchurl
+, makeWrapper
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "quarto";
+  version = "1.0.38";
+
+  src = fetchurl {
+    url = "https://github.com/quarto-dev/quarto-cli/releases/download/v${finalAttrs.version}/quarto-${finalAttrs.version}-linux-amd64.tar.gz";
```

```suggestion
stdenv.mkDerivation rec {
  pname = "quarto";
  version = "1.0.38";

  src = fetchurl {
    url = "https://github.com/quarto-dev/quarto-cli/releases/download/v${version}/quarto-${version}-linux-amd64.tar.gz";
```
finalAttrs does not work with fixed output derivations.

---

What is a "fixed output derivation"?

---

a fetcher with a hash for example. If you change the input arguments and the hash stays the same, nix just re-uses the old output.

---

Well, changing the input without changing the hash sounds like a not correctly working hash.
Anyway, if I change the source of the downloader without changing the hash does not work with the current implementation. The builder will interrupt the process as expected. What do you mean here in specific?

---

> Well, changing the input without changing the hash sounds like a not correctly working hash.

no, this is exactly how it should work for a fixed output derivation and why finalAttrs is not useful here and should be removed. If src should be changed people should use overrideAttrs and overwrite src there.